src/fides/api/models/privacy_preference.py

```diff
@@ -20,10 +20,19 @@
 
 
 class RequestOrigin(Enum):
-    # Not at the db level due to being subject to change.
-    # Only add here, do not remove
+    """
+    We currently extract the RequestOrigin from the Privacy Experience
+    Config ComponentType when saving, so RequestOrigin needs to be a
+    superset of ComponentType.
+
+    Not at the db level due to being subject to change.
+    Only add here, do not remove
+    """
+
     privacy_center = "privacy_center"
-    overlay = "overlay"
+    overlay = "overlay"  # DEPRECATED. DO NOT REMOVE.
+    modal = "modal"
+    banner_and_modal = "banner_and_modal"
```

Comment:
New modal and banner_and_modal types added. Overlay is deprecated but cannot be removed for backwards compatibility.